plugins/ipam/dhcp/lease.go

```diff
@@ -344,6 +362,51 @@ func (l *DHCPLease) maintain() {
 	}
 }
 
+func (l *DHCPLease) checkLinkExistsWithBackoff(ctx context.Context) (bool, error) {
+	checkFunc := func() (bool, error) {
+		_, err := netlink.LinkByName(l.link.Attrs().Name)
+		if err != nil {
+			log.Printf("!bang: Error checking network link '%s': %v, type: %v", l.link.Attrs().Name, err, reflect.TypeOf(err))
+			var linkNotFoundErr *netlink.LinkNotFoundError
+			if errors.As(err, &linkNotFoundErr) {
```

Comment:
This wasn't evaluating to true even though I was seeing logs as:

```
2025/01/21 21:36:46 Error checking network link 'net1': Link not found, type: netlink.LinkNotFoundError
```

I'll remove the reflect package and also the lines marked with `!bang` once I get it sorted.